components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/dao/impl/CacheBackedAuthenticatorMgtDAO.java

```diff
@@ -53,10 +58,10 @@ public UserDefinedLocalAuthenticatorConfig addUserDefinedLocalAuthenticator(
 
         AuthenticatorCacheKey cacheKey = new AuthenticatorCacheKey(authenticatorConfig.getName());
         authenticatorCache.addToCache(cacheKey, new AuthenticatorCacheEntry(createdConfig), tenantId);
-        if (LOG.isDebugEnabled()) {
-            LOG.debug(String.format(
-                    "Adding cache entry for newly created authenticator %s.", authenticatorConfig.getName()));
-        }
+        LOG.debug("Adding cache entry for newly created authenticator " + authenticatorConfig.getName());
+        userDefinedLocalAuthenticatorsCache.clearCacheEntry(new UserDefinedLocalAuthenticatorsCacheKey(tenantId),
+                tenantId);
+        LOG.debug("Delete Cache entry for all user defined local authenticators of tenant id: " + tenantId);
```

Comment:
And this should be 'Deleted'

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/6554/commits/c8f22853da6a60ae52a5a39fb0427f7b930064da